src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultConverters.cs

```diff
@@ -0,0 +1,303 @@
+                {
+                    // Then declaring assembly
+                    attr = options.GetAttributes<JsonPropertyValueAttribute>(parentClassType.Assembly).FirstOrDefault();
+
+                    if (attr == null || (value = selector(attr)) == default)
+                    {
+                        // Then global
+                        attr = options.GetAttributes<JsonPropertyValueAttribute>(JsonSerializerOptions.GlobalAttributesProvider).FirstOrDefault();
+
+                        if (attr == null)
+                        {
+                            value = default;
+                        }
+                        else
+                        {
+                            value = selector(attr);
+                        }
+                    }
+                }
+            }
+
+            return value;
+        }
+
+        public static JsonEnumerableConverter GetEnumerableConverter(
+            Type parentClassType,
+            PropertyInfo propertyInfo,
+            Type propertyType,
+            JsonSerializerOptions options)
+        {
+            Type enumerableType;
+            if (propertyType.IsGenericType)
+            {
+                enumerableType = propertyType.GetGenericTypeDefinition();
+            }
+            else if (propertyType.IsArray)
+            {
+                enumerableType = typeof(Array);
+            }
+            else
+            {
+                enumerableType = propertyType;
+            }
+
+            JsonEnumerableConverterAttribute attr = null;
+            if (propertyInfo != null)
+            {
+                // Use Property first
+                attr = options.GetAttributes<JsonEnumerableConverterAttribute>(propertyInfo).Where(a => a.EnumerableType == enumerableType).FirstOrDefault();
+            }
+
+            if (attr == null)
+            {
+                // Then class type
+                attr = options.GetAttributes<JsonEnumerableConverterAttribute>(parentClassType, inherit: true).Where(a => a.EnumerableType == enumerableType).FirstOrDefault();
+
+                if (attr == null)
+                {
+                    // Then declaring assembly
+                    attr = options.GetAttributes<JsonEnumerableConverterAttribute>(parentClassType.Assembly).Where(a => a.EnumerableType == enumerableType).FirstOrDefault();
+
+                    if (attr == null)
+                    {
+                        // Then global
+                        attr = options.GetAttributes<JsonEnumerableConverterAttribute>(JsonSerializerOptions.GlobalAttributesProvider).Where(a => a.EnumerableType == enumerableType).FirstOrDefault();
+
+                        // Then default
+                        if (enumerableType == typeof(Array))
+                        {
+                            attr = s_arrayConverterAttibute;
+                        }
+                    }
+                }
+            }
+
+            return attr?.CreateConverter();
```

Comment:
No it could be null if there is no attribute for the enumerable type - basically this code allows you to write a custom converter to populate something like a "stack" or immutable collection type.